server/repository/src/mock/stock_line.rs

```diff
@@ -18,6 +18,7 @@ pub fn mock_stock_line_a() -> StockLineRow {
         on_hold: false,
         note: None,
         supplier_id: Some(String::from("name_store_b")),
+        barcode_id: None,
```

Comment:
yeah, I know. someone should create a Default for this.

Reply:
Not require change at all
there is a script for this btw (although kind of prefer the spreading he default now rather then inline_init):
https://github.com/openmsupply/open-msupply/blob/baae025c0a78dda0993b60bb5a57a08f6d3d1653/server/util/src/to_inline_init.js#L4

from server dir, ran: `node util/src/to_inline_init.js repository/src/mock/stock_line.rs StockLineRow`. then opened   `repository/src/mock/stock_line.rs`, and saved it (for formatting) and imported inline_init with `cmd + .`, and got this:

```diff
diff --git a/server/repository/src/mock/stock_line.rs b/server/repository/src/mock/stock_line.rs
index 25dc41d6b..43f2adca9 100644
--- a/server/repository/src/mock/stock_line.rs
+++ b/server/repository/src/mock/stock_line.rs
@@ -1,45 +1,31 @@
 use chrono::NaiveDate;
+use util::inline_init;
 
 use crate::StockLineRow;
 
 pub fn mock_stock_line_a() -> StockLineRow {
-    StockLineRow {
-        id: String::from("item_a_line_a"),
-        item_id: String::from("item_a"),
-        location_id: None,
-        store_id: String::from("store_a"),
-        batch: None,
-        available_number_of_packs: 30.0,
-        pack_size: 1,
-        cost_price_per_pack: 0.0,
-        sell_price_per_pack: 0.0,
-        total_number_of_packs: 40.0,
-        expiry_date: None,
-        on_hold: false,
-        note: None,
-        supplier_id: Some(String::from("name_store_b")),
-        barcode_id: None,
-    }
+    inline_init(|r: &mut StockLineRow| {
+        r.id = String::from("item_a_line_a");
+        r.item_id = String::from("item_a");
+        r.store_id = String::from("store_a");
+        r.available_number_of_packs = 30.0;
+        r.pack_size = 1;
+        r.total_number_of_packs = 40.0;
+        r.supplier_id = Some(String::from("name_store_b"));
+    })
 }
 
 pub fn mock_stock_line_b() -> StockLineRow {
-    StockLineRow {
-        id: String::from("item_a_line_b"),
-        item_id: String::from("item_a"),
-        location_id: None,
-        store_id: String::from("store_a"),
-        batch: Some(String::from("item_a_batch_b")),
-        available_number_of_packs: 20.0,
-        pack_size: 1,
-        cost_price_per_pack: 0.0,
-        sell_price_per_pack: 0.0,
-        total_number_of_packs: 30.0,
-        expiry_date: None,
-        on_hold: false,
-        note: None,
-        supplier_id: Some(String::from("name_store_b")),
-        barcode_id: None,
-    }
+    inline_init(|r: &mut StockLineRow| {
+        r.id = String::from("item_a_line_b");
+        r.item_id = String::from("item_a");
+        r.store_id = String::from("store_a");
+        r.batch = Some(String::from("item_a_batch_b"));
+        r.available_number_of_packs = 20.0;
+        r.pack_size = 1;
+        r.total_number_of_packs = 30.0;
+        r.supplier_id = Some(String::from("name_store_b"));
+    })
 }
 
 pub fn mock_item_a_lines() -> Vec<StockLineRow> {
@@ -51,249 +37,182 @@ pub fn mock_item_a_lines() -> Vec<StockLineRow> {
 }
 
 pub fn mock_item_b_lines() -> Vec<StockLineRow> {
-    let mock_item_b_line_a: StockLineRow = StockLineRow {
-        id: String::from("item_b_line_a"),
-        item_id: String::from("item_b"),
-        location_id: None,
-        store_id: String::from("store_b"),
-        batch: Some(String::from("item_b_batch_a")),
-        available_number_of_packs: 3.0,
-        pack_size: 1,
-        cost_price_per_pack: 0.0,
-        sell_price_per_pack: 0.0,
-        total_number_of_packs: 30.0,
-        expiry_date: None,
-        on_hold: false,
-        note: None,
-        supplier_id: Some(String::from("name_store_b")),
-        barcode_id: None,
-    };
-
-    let mock_item_b_line_b: StockLineRow = StockLineRow {
-        id: String::from("item_b_line_b"),
-        item_id: String::from("item_b"),
-        location_id: None,
-        store_id: String::from("store_b"),
-        batch: Some(String::from("item_b_batch_b")),
-        available_number_of_packs: 4.0,
-        pack_size: 1,
-        cost_price_per_pack: 0.0,
-        sell_price_per_pack: 0.0,
-        total_number_of_packs: 25.0,
-        expiry_date: None,
-        on_hold: false,
-        note: None,
-        supplier_id: Some(String::from("name_store_b")),
-        barcode_id: None,
-    };
+    let mock_item_b_line_a: StockLineRow = inline_init(|r: &mut StockLineRow| {
+        r.id = String::from("item_b_line_a");
+        r.item_id = String::from("item_b");
+        r.store_id = String::from("store_b");
+        r.batch = Some(String::from("item_b_batch_a"));
+        r.available_number_of_packs = 3.0;
+        r.pack_size = 1;
+        r.total_number_of_packs = 30.0;
+        r.supplier_id = Some(String::from("name_store_b"));
+    });
+
+    let mock_item_b_line_b: StockLineRow = inline_init(|r: &mut StockLineRow| {
+        r.id = String::from("item_b_line_b");
+        r.item_id = String::from("item_b");
+        r.store_id = String::from("store_b");
+        r.batch = Some(String::from("item_b_batch_b"));
+        r.available_number_of_packs = 4.0;
+        r.pack_size = 1;
+        r.total_number_of_packs = 25.0;
+        r.supplier_id = Some(String::from("name_store_b"));
+    });
 
     vec![mock_item_b_line_a, mock_item_b_line_b]
 }
 
 pub fn mock_item_c_lines() -> Vec<StockLineRow> {
-    let mock_item_c_line_a: StockLineRow = StockLineRow {
-        id: String::from("item_c_line_a"),
-        item_id: String::from("item_c"),
-        location_id: None,
-        store_id: String::from("store_c"),
-        batch: Some(String::from("item_c_batch_a")),
-        available_number_of_packs: 5.0,
-        pack_size: 1,
-        cost_price_per_pack: 12.0,
-        sell_price_per_pack: 15.0,
-        total_number_of_packs: 1.0,
-        expiry_date: None,
-        on_hold: false,
-        note: Some("stock line note".to_owned()),
-        supplier_id: Some(String::from("name_store_b")),
-        barcode_id: None,
-    };
-
-    let mock_item_c_line_b: StockLineRow = StockLineRow {
-        id: String::from("item_c_line_b"),
-        item_id: String::from("item_c"),
-        location_id: None,
-        store_id: String::from("store_c"),
-        batch: Some(String::from("item_c_batch_b")),
-        available_number_of_packs: 6.0,
-        pack_size: 1,
-        cost_price_per_pack: 0.0,
-        sell_price_per_pack: 0.0,
-        total_number_of_packs: 1.0,
-        expiry_date: None,
-        on_hold: false,
-        note: None,
-        supplier_id: Some(String::from("name_store_b")),
-        barcode_id: None,
-    };
+    let mock_item_c_line_a: StockLineRow = inline_init(|r: &mut StockLineRow| {
+        r.id = String::from("item_c_line_a");
+        r.item_id = String::from("item_c");
+        r.store_id = String::from("store_c");
+        r.batch = Some(String::from("item_c_batch_a"));
+        r.available_number_of_packs = 5.0;
+        r.pack_size = 1;
+        r.cost_price_per_pack = 12.0;
+        r.sell_price_per_pack = 15.0;
+        r.total_number_of_packs = 1.0;
+        r.note = Some("stock line note".to_owned());
+        r.supplier_id = Some(String::from("name_store_b"));
+    });
+
+    let mock_item_c_line_b: StockLineRow = inline_init(|r: &mut StockLineRow| {
+        r.id = String::from("item_c_line_b");
+        r.item_id = String::from("item_c");
+        r.store_id = String::from("store_c");
+        r.batch = Some(String::from("item_c_batch_b"));
+        r.available_number_of_packs = 6.0;
+        r.pack_size = 1;
+        r.total_number_of_packs = 1.0;
+        r.supplier_id = Some(String::from("name_store_b"));
+    });
 
     vec![mock_item_c_line_a, mock_item_c_line_b]
 }
 
 pub fn mock_stock_line_si_d() -> Vec<StockLineRow> {
-    let mock_stock_line_si_d_siline_a: StockLineRow = StockLineRow {
-        id: String::from("stock_line_si_d_siline_a"),
-        item_id: String::from("item_a"),
-        location_id: None,
-        store_id: String::from("store_a"),
-        batch: Some(String::from("item_a_si_d_siline_a")),
-        available_number_of_packs: 7.0,
-        pack_size: 1,
-        cost_price_per_pack: 2.0,
-        sell_price_per_pack: 18.0,
-        total_number_of_packs: 7.0,
-        expiry_date: None,
-        on_hold: false,
-        note: None,
-        supplier_id: Some(String::from("name_store_b")),
-        barcode_id: None,
-    };
-
-    let mock_stock_line_si_d_siline_b: StockLineRow = StockLineRow {
-        id: String::from("stock_line_si_d_siline_b"),
-        item_id: String::from("item_b"),
-        location_id: None,
-        store_id: String::from("store_a"),
-        batch: Some(String::from("item_b_si_c_siline_d")),
-        available_number_of_packs: 2.0,
-        pack_size: 3,
-        cost_price_per_pack: 45.0,
-        sell_price_per_pack: 34.0,
-        total_number_of_packs: 2.0,
-        expiry_date: Some(NaiveDate::from_ymd_opt(2020, 8, 11).unwrap()),
-        on_hold: false,
-        note: None,
-        supplier_id: Some(String::from("name_store_b")),
-        barcode_id: None,
-    };
+    let mock_stock_line_si_d_siline_a: StockLineRow = inline_init(|r: &mut StockLineRow| {
+        r.id = String::from("stock_line_si_d_siline_a");
+        r.item_id = String::from("item_a");
+        r.store_id = String::from("store_a");
+        r.batch = Some(String::from("item_a_si_d_siline_a"));
+        r.available_number_of_packs = 7.0;
+        r.pack_size = 1;
+        r.cost_price_per_pack = 2.0;
+        r.sell_price_per_pack = 18.0;
+        r.total_number_of_packs = 7.0;
+        r.supplier_id = Some(String::from("name_store_b"));
+    });
+
+    let mock_stock_line_si_d_siline_b: StockLineRow = inline_init(|r: &mut StockLineRow| {
+        r.id = String::from("stock_line_si_d_siline_b");
+        r.item_id = String::from("item_b");
+        r.store_id = String::from("store_a");
+        r.batch = Some(String::from("item_b_si_c_siline_d"));
+        r.available_number_of_packs = 2.0;
+        r.pack_size = 3;
+        r.cost_price_per_pack = 45.0;
+        r.sell_price_per_pack = 34.0;
+        r.total_number_of_packs = 2.0;
+        r.expiry_date = Some(NaiveDate::from_ymd_opt(2020, 8, 11).unwrap());
+        r.supplier_id = Some(String::from("name_store_b"));
+    });
 
     vec![mock_stock_line_si_d_siline_a, mock_stock_line_si_d_siline_b]
 }
 
 pub fn mock_stock_line_ci_c() -> Vec<StockLineRow> {
-    let mock_stock_line_ci_c_siline_a: StockLineRow = StockLineRow {
-        id: String::from("stock_line_ci_c_siline_a"),
-        item_id: String::from("item_a"),
-        location_id: None,
-        store_id: String::from("store_a"),
-        batch: Some(String::from("item_a_ci_c_siline_a")),
-        available_number_of_packs: 5.0,
-        pack_size: 3,
-        cost_price_per_pack: 8.0,
-        sell_price_per_pack: 9.0,
-        total_number_of_packs: 8.0,
-        expiry_date: Some(NaiveDate::from_ymd_opt(2020, 1, 4).unwrap()),
-        on_hold: false,
-        note: None,
-        supplier_id: Some(String::from("name_store_b")),
-        barcode_id: None,
-    };
-
-    let mock_stock_line_ci_c_siline_b: StockLineRow = StockLineRow {
-        id: String::from("stock_line_ci_c_siline_b"),
-        item_id: String::from("item_b"),
-        location_id: None,
-        store_id: String::from("store_a"),
-        batch: None,
-        available_number_of_packs: 20.0,
-        pack_size: 7,
-        cost_price_per_pack: 54.0,
-        sell_price_per_pack: 34.0,
-        total_number_of_packs: 21.0,
-        expiry_date: Some(NaiveDate::from_ymd_opt(2020, 3, 23).unwrap()),
-        on_hold: false,
-        note: None,
-        supplier_id: Some(String::from("name_store_b")),
-        barcode_id: None,
-    };
+    let mock_stock_line_ci_c_siline_a: StockLineRow = inline_init(|r: &mut StockLineRow| {
+        r.id = String::from("stock_line_ci_c_siline_a");
+        r.item_id = String::from("item_a");
+        r.store_id = String::from("store_a");
+        r.batch = Some(String::from("item_a_ci_c_siline_a"));
+        r.available_number_of_packs = 5.0;
+        r.pack_size = 3;
+        r.cost_price_per_pack = 8.0;
+        r.sell_price_per_pack = 9.0;
+        r.total_number_of_packs = 8.0;
+        r.expiry_date = Some(NaiveDate::from_ymd_opt(2020, 1, 4).unwrap());
+        r.supplier_id = Some(String::from("name_store_b"));
+    });
+
+    let mock_stock_line_ci_c_siline_b: StockLineRow = inline_init(|r: &mut StockLineRow| {
+        r.id = String::from("stock_line_ci_c_siline_b");
+        r.item_id = String::from("item_b");
+        r.store_id = String::from("store_a");
+        r.available_number_of_packs = 20.0;
+        r.pack_size = 7;
+        r.cost_price_per_pack = 54.0;
+        r.sell_price_per_pack = 34.0;
+        r.total_number_of_packs = 21.0;
+        r.expiry_date = Some(NaiveDate::from_ymd_opt(2020, 3, 23).unwrap());
+        r.supplier_id = Some(String::from("name_store_b"));
+    });
 
     vec![mock_stock_line_ci_c_siline_a, mock_stock_line_ci_c_siline_b]
 }
 
 pub fn mock_stock_line_ci_d() -> Vec<StockLineRow> {
-    let mock_stock_line_ci_d_siline_a: StockLineRow = StockLineRow {
-        id: String::from("stock_line_ci_d_siline_a"),
-        item_id: String::from("item_a"),
-        location_id: None,
-        store_id: String::from("store_a"),
-        batch: Some(String::from("item_a_ci_d_siline_a")),
-        available_number_of_packs: 10.0,
-        pack_size: 1,
-        cost_price_per_pack: 10.0,
-        sell_price_per_pack: 11.0,
-        total_number_of_packs: 10.0,
-        expiry_date: Some(NaiveDate::from_ymd_opt(2020, 1, 4).unwrap()),
-        on_hold: false,
-        note: None,
-        supplier_id: Some(String::from("name_store_b")),
-        barcode_id: None,
-    };
+    let mock_stock_line_ci_d_siline_a: StockLineRow = inline_init(|r: &mut StockLineRow| {
+        r.id = String::from("stock_line_ci_d_siline_a");
+        r.item_id = String::from("item_a");
+        r.store_id = String::from("store_a");
+        r.batch = Some(String::from("item_a_ci_d_siline_a"));
+        r.available_number_of_packs = 10.0;
+        r.pack_size = 1;
+        r.cost_price_per_pack = 10.0;
+        r.sell_price_per_pack = 11.0;
+        r.total_number_of_packs = 10.0;
+        r.expiry_date = Some(NaiveDate::from_ymd_opt(2020, 1, 4).unwrap());
+        r.supplier_id = Some(String::from("name_store_b"));
+    });
 
     vec![mock_stock_line_ci_d_siline_a]
 }
 
 pub fn mock_item_query_test1() -> Vec<StockLineRow> {
-    let mock_item_query_test1: StockLineRow = StockLineRow {
-        id: "item_query_test1".to_owned(),
-        item_id: "item_query_test1".to_owned(),
-        location_id: None,
-        store_id: "store_a".to_owned(),
-        batch: None,
-        available_number_of_packs: 3.0,
-        pack_size: 12,
-        cost_price_per_pack: 2.0,
-        sell_price_per_pack: 11.0,
-        total_number_of_packs: 3.0,
-        expiry_date: None,
-        on_hold: false,
-        note: None,
-        supplier_id: Some(String::from("name_store_b")),
-        barcode_id: None,
-    };
+    let mock_item_query_test1: StockLineRow = inline_init(|r: &mut StockLineRow| {
+        r.id = "item_query_test1".to_owned();
+        r.item_id = "item_query_test1".to_owned();
+        r.store_id = "store_a".to_owned();
+        r.available_number_of_packs = 3.0;
+        r.pack_size = 12;
+        r.cost_price_per_pack = 2.0;
+        r.sell_price_per_pack = 11.0;
+        r.total_number_of_packs = 3.0;
+        r.supplier_id = Some(String::from("name_store_b"));
+    });
 
     vec![mock_item_query_test1]
 }
 
 pub fn mock_stock_line_on_hold() -> Vec<StockLineRow> {
-    let mock_stock_line_on_hold: StockLineRow = StockLineRow {
-        id: "stock_line_on_hold".to_owned(),
-        item_id: "item_c".to_owned(),
-        location_id: None,
-        store_id: "store_a".to_owned(),
-        batch: None,
-        available_number_of_packs: 100.0,
-        pack_size: 1,
-        cost_price_per_pack: 0.0,
-        sell_price_per_pack: 0.0,
-        total_number_of_packs: 100.0,
-        expiry_date: None,
-        on_hold: true,
-        note: None,
-        supplier_id: Some(String::from("name_store_b")),
-        barcode_id: None,
-    };
+    let mock_stock_line_on_hold: StockLineRow = inline_init(|r: &mut StockLineRow| {
+        r.id = "stock_line_on_hold".to_owned();
+        r.item_id = "item_c".to_owned();
+        r.store_id = "store_a".to_owned();
+        r.available_number_of_packs = 100.0;
+        r.pack_size = 1;
+        r.total_number_of_packs = 100.0;
+        r.on_hold = true;
+        r.supplier_id = Some(String::from("name_store_b"));
+    });
 
     vec![mock_stock_line_on_hold]
 }
 
 pub fn mock_stock_line_location_is_on_hold() -> Vec<StockLineRow> {
-    let mock_stock_line_location_is_on_hold: StockLineRow = StockLineRow {
-        id: "stock_line_location_is_on_hold".to_owned(),
-        item_id: "item_c".to_owned(),
-        location_id: Some("location_on_hold".to_owned()),
-        store_id: "store_a".to_owned(),
-        batch: None,
-        available_number_of_packs: 100.0,
-        pack_size: 1,
-        cost_price_per_pack: 0.0,
-        sell_price_per_pack: 0.0,
-        total_number_of_packs: 100.0,
-        expiry_date: None,
-        on_hold: false,
-        note: None,
-        supplier_id: Some(String::from("name_store_b")),
-        barcode_id: None,
-    };
+    let mock_stock_line_location_is_on_hold: StockLineRow = inline_init(|r: &mut StockLineRow| {
+        r.id = "stock_line_location_is_on_hold".to_owned();
+        r.item_id = "item_c".to_owned();
+        r.location_id = Some("location_on_hold".to_owned());
+        r.store_id = "store_a".to_owned();
+        r.available_number_of_packs = 100.0;
+        r.pack_size = 1;
+        r.total_number_of_packs = 100.0;
+        r.supplier_id = Some(String::from("name_store_b"));
+    });
 
     vec![mock_stock_line_location_is_on_hold]
 }
```

Reply:
gosh. impressive!
I do like the use of defaults, it makes it clear what is being extended / changed from the default values. Will update with the inline init in this case.. too good not to.